python/lsst/daf/butler/datastore/file_templates.py

```diff
@@ -406,25 +414,26 @@ def grouped_fields(self) -> tuple[dict[str, set[str]], dict[str, set[str]]]:
             "parent": set(),
         }
 
-        for _, field_name, format_spec, _ in parts:
-            if field_name is not None and format_spec is not None:
-                subfield = None
-                key = "standard"
-                if field_name in self.specialFields:
-                    key = "special"
-                elif "." in field_name:
-                    # This needs to be added twice.
-                    subfield = field_name
-                    key = "parent"
-                    field_name, _ = field_name.split(".")
-
-                if "?" in format_spec:
-                    target = grouped_optional
-                else:
-                    target = grouped
-                target[key].add(field_name)
-                if subfield is not None:
-                    target["subfield"].add(subfield)
+        for _, field_names, format_spec, _ in parts:
+            if field_names is not None and format_spec is not None:
+                for field_name in field_names.split("|"):  # Treat alternates as equals.
+                    subfield = None
+                    key = "standard"
+                    if field_name in self.specialFields:
+                        key = "special"
+                    elif "." in field_name:
+                        # This needs to be added twice.
+                        subfield = field_name
+                        key = "parent"
+                        field_name, _ = field_name.split(".")
```

Comment:
I think the logic would be clearer if you moved the `if "?"` block that sets `target` above this one, so you don't need the separate `if subfield` later.